Update mark notification as read method

flamisz opened the pull request:
### Description:

As in title.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

flamisz committed:
update mark notification as read method

sgiehl reviewed (changes requested):
@flamisz  Did you test that locally? For me the XHR request is actually never sent, as the notificationId is always null. Guess it needs to be passed to `markNotificationAsRead` from the html template instead, as it isn't available in the controller.

tsteur commented:
btw @sgiehl @flamisz not sure what you exactly meant in the comment but if the request isn't sent currently that might be because of https://github.com/matomo-org/matomo/issues/15575#issuecomment-822897559 which may need to be fixed in order to test this issue :)

tsteur commented:
it might be totally unrelated though so feel free to ignore, only thought to mention it in case it's related

flamisz commented:
> @flamisz Did you test that locally? For me the XHR request is actually never sent, as the notificationId is always null. Guess it needs to be passed to `markNotificationAsRead` from the html template instead, as it isn't available in the controller.

@tsteur @sgiehl I realizes something is not ok with the notification, but I thought I just don't know how to use it properly. I was able to test it tho. I created permanent notification (hardcoded somewhere into a plugin) and used an API call to mark as readed. Before the PR I was able to mark as readed so it disappeared for the next request. With this modifications I needed a token to do that.

tsteur commented:
👍  we'd just need to double check that we also send the token in the actual request from the notification JS otherwise it won't work basically (but it's not working there currently anyway so could also ignore it and not test it)

flamisz commented on plugins/CoreHome/angularjs/notification/notification.controller.js:
```diff
@@ -20,6 +20,7 @@
                 return;
             }
 
+            piwikApi.withTokenInUrl();
```
> 👍 we'd just need to double check that we also send the token in the actual request from the notification JS otherwise it won't work basically (but it's not working there currently anyway so could also ignore it and not test it)

@tsteur this pr contains this modification here ⬆️

sgiehl committed:
Ensure notificationId is passed correctly to markNotificationAsRead

sgiehl commented:
@flamisz while reviewing I actually saw why the notification xhr didn't work properly. I've pushed a commit to this branch to fix that. So maybe this then also fixes #15575

flamisz commented:
> @flamisz while reviewing I actually saw why the notification xhr didn't work properly. I've pushed a commit to this branch to fix that. So maybe this then also fixes #15575

thanks @sgiehl, this makes sense. I just wasn't sure if it's a bug or my lack of knowledge 😄

sgiehl committed:
Adds notification UI tests

sgiehl commented:
@flamisz I've rebased the branch and added some new UI tests to check that closing the notifications work as expected. Feel free to have a look at the UI tests and merge the PR if they look good to you.

flamisz commented:
> @flamisz I've rebased the branch and added some new UI tests to check that closing the notifications work as expected. Feel free to have a look at the UI tests and merge the PR if they look good to you.

Thanks, @sgiehl, looking good 👍  
Probably wait with the merge until we have the new release.

sgiehl commented:
Might maybe be worth to include that in 4.3.0 maybe, as it fixes the uncloseable notification issue. ping @tsteur
